src/externalDevice/BleChessProtocol.ts

```diff
@@ -41,242 +154,167 @@ abstract class BleChessState extends BaseState {
 
 class Init extends BleChessState {
   onEnter() {
-    this.transitionTo(new CheckFeatureMsg)
+    const checkVariants = new CheckIteration(createIterator(this.getVariants()), Command.Variant, new Initialized)
+    const checkFeatures = new CheckIteration(createIterator(this.getFeatures()), Command.Feature, checkVariants)
+    this.transitionTo(checkFeatures)
   }
 }
 
-class CheckFeatureMsg extends BleChessState {
+class CheckIteration extends BleChessState {
```

Comment:
maybe SupportListChecker?

Reply:
Proposal is `CheckSupportsIteration` and `createSupportsIteration`